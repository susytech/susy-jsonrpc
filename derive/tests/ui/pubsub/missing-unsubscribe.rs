#[macro_use]
extern crate susy_jsonrpc_derive;
extern crate susy_jsonrpc_core;
extern crate susy_jsonrpc_pubsub;

#[rpc]
pub trait Rpc {
	type Metadata;

	/// Hello subscription
	#[pubsub(subscription = "hello", subscribe, name = "hello_subscribe", alias("hello_sub"))]
	fn subscribe(&self, Self::Metadata, typed::Subscriber<String>, u64);

	// note that the unsubscribe method is missing
}

fn main() {}
