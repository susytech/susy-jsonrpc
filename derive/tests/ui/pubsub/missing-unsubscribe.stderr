error: custom attribute panicked
 --> $DIR/missing-unsubscribe.rs:6:1
  |
6 | #[rpc]
  | ^^^^^^
  |
  = help: message: [rpc] encountered error: subscription 'hello'. Can't find unsubscribe method, expected a method annotated with `unsubscribe` e.g. `#[pubsub(subscription = "hello", unsubscribe, name = "hello_unsubscribe")]`

error: aborting due to previous error

